src/util.rs

```diff
@@ -336,15 +334,29 @@ mod test {
     #[cfg(feature = "update-notifications")]
     #[test]
     pub fn compare_ver_test() {
-        assert_eq!(compare_versions("0.1.0", "0.1.0"), Some(true));
-        assert_eq!(compare_versions("1.1.0", "0.1.0"), Some(true));
-        assert_eq!(compare_versions("1.0.0", "0.2.5"), Some(true));
-        assert_eq!(compare_versions("1.0.0", "2.2.5"), Some(false));
-        assert_eq!(compare_versions("1.0.0", "2.0.5"), Some(false));
-        assert_eq!(compare_versions("1.1.0", "2.0.5"), Some(false));
-        assert_eq!(compare_versions("1.1.6", "2.0.0"), Some(false));
-        assert_eq!(compare_versions("0.1.1", "0.1.0"), Some(true));
-        assert_eq!(compare_versions("0.1.1", "0.2.0"), Some(false));
+        use super::VersionComparison::*;
+        assert_eq!(compare_versions("0.1.0", "0.1.0").unwrap(), NewIsEqual);
+        assert_eq!(compare_versions("1.1.0", "0.1.0").unwrap(), NewIsLesser);
+        assert_eq!(compare_versions("1.0.0", "0.2.5").unwrap(), NewIsLesser);
+        assert_eq!(compare_versions("1.0.0", "2.2.5").unwrap(), NewIsGreater);
+        assert_eq!(compare_versions("1.0.0", "2.0.5").unwrap(), NewIsGreater);
+        assert_eq!(compare_versions("1.1.0", "2.0.5").unwrap(), NewIsGreater);
+        assert_eq!(compare_versions("1.1.6", "2.0.0").unwrap(), NewIsGreater);
+        assert_eq!(compare_versions("0.1.1", "0.1.0").unwrap(), NewIsLesser);
+        assert_eq!(compare_versions("0.1.1", "0.2.0").unwrap(), NewIsGreater);
+
+        assert_eq!(compare_versions("v0.1.0", "v0.1.0").unwrap(), NewIsEqual);
+        assert_eq!(compare_versions("v1.1.0", "v0.1.0").unwrap(), NewIsLesser);
+        assert_eq!(compare_versions("v1.1.6", "v2.0.0").unwrap(), NewIsGreater);
+
+        assert_eq!(compare_versions("0.1.0", "v0.1.0").unwrap(), NewIsEqual);
+        assert_eq!(compare_versions("1.1.0", "v0.1.0").unwrap(), NewIsLesser);
+        assert_eq!(compare_versions("1.1.6", "v2.0.0").unwrap(), NewIsGreater);
+
+        assert_eq!(compare_versions("v0.1.0", "0.1.0").unwrap(), NewIsEqual);
+        assert_eq!(compare_versions("v1.1.0", "0.1.0").unwrap(), NewIsLesser);
+        assert_eq!(compare_versions("v1.1.6", "2.0.0").unwrap(), NewIsGreater);
+        assert_eq!(compare_versions("3.0.2", "v3.0.2").unwrap(), NewIsEqual);
     }
```

Comment:
Should have some tests with a `v` prefix so the stripping is tested.

Reply:
It does have tests with `v` stripping

Reply:
Whoops, I'm blind, sorry.